pkg/agent/controller/bgp/controller_test.go

```diff
@@ -2166,37 +2166,103 @@ func TestGetBGPPeerStatus(t *testing.T) {
 	ctx := context.Background()
 	testCases := []struct {
 		name                  string
+		ipv4Peers             bool
+		ipv6Peers             bool
 		existingState         *bgpPolicyState
 		expectedCalls         func(mockBGPServer *bgptest.MockInterfaceMockRecorder)
 		expectedBgpPeerStatus []bgp.PeerStatus
 		expectedErr           error
 	}{
 		{
-			name:          "bgpPolicyState exists",
+			name:          "get ipv4 bgp peers",
+			ipv4Peers:     true,
 			existingState: &bgpPolicyState{},
 			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
 				mockBGPServer.GetPeers(ctx).Return([]bgp.PeerStatus{
 					{
-						Address:      "192.168.77.200",
-						ASN:          65001,
+						Address:      ipv4Peer1Addr,
+						ASN:          peer1ASN,
 						SessionState: bgp.SessionEstablished,
 					},
 					{
-						Address:      "192.168.77.201",
-						ASN:          65002,
+						Address:      ipv4Peer2Addr,
+						ASN:          peer2ASN,
 						SessionState: bgp.SessionActive,
 					},
 				}, nil)
 			},
 			expectedBgpPeerStatus: []bgp.PeerStatus{
 				{
-					Address:      "192.168.77.200",
-					ASN:          65001,
+					Address:      ipv4Peer1Addr,
+					ASN:          peer1ASN,
 					SessionState: bgp.SessionEstablished,
 				},
 				{
-					Address:      "192.168.77.201",
-					ASN:          65002,
+					Address:      ipv4Peer2Addr,
+					ASN:          peer2ASN,
+					SessionState: bgp.SessionActive,
+				},
+			},
+		},
+		{
+			name:          "get ipv6 bgp peers",
+			ipv6Peers:     true,
+			existingState: &bgpPolicyState{},
+			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
+				mockBGPServer.GetPeers(ctx).Return([]bgp.PeerStatus{
+					{
+						Address:      ipv6Peer1Addr,
+						ASN:          peer1ASN,
+						SessionState: bgp.SessionEstablished,
+					},
+					{
+						Address:      ipv6Peer2Addr,
+						ASN:          peer2ASN,
+						SessionState: bgp.SessionActive,
+					},
+				}, nil)
+			},
+			expectedBgpPeerStatus: []bgp.PeerStatus{
+				{
+					Address:      ipv6Peer1Addr,
+					ASN:          peer1ASN,
+					SessionState: bgp.SessionEstablished,
+				},
+				{
+					Address:      ipv6Peer2Addr,
+					ASN:          peer2ASN,
+					SessionState: bgp.SessionActive,
+				},
+			},
+		},
+		{
+			name:          "get all bgp peers",
+			ipv4Peers:     true,
+			ipv6Peers:     true,
+			existingState: &bgpPolicyState{},
+			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
+				mockBGPServer.GetPeers(ctx).Return([]bgp.PeerStatus{
+					{
```

Comment:
can you define 4 variables of type `bgp.PeerStatus` to avoid the repetition: `ipv4Peer1Status`, `ipv4Peer2Status`, `ipv6Peer1Status`, `ipv6Peer2Status`